File: AlternatingCharacters/AlternatingCharacters.lua
-- Complete the alternatingCharacters function below.
function alternatingCharacters(s)

    local deletion = 0;

    for i=1, #s do
        if (i ~= #s and s:sub(i,i) == s:sub(i+1, i+1)) then
            deletion = deletion + 1;
        end
    end

    return deletion
end

local fptr = io.open(os.getenv("OUTPUT_PATH"), "w")

local q = io.stdin:read("*n", "*l")

for qitr = 1, q do
    local s = io.stdin:read("*l")

    local result = alternatingCharacters(s)

    fptr:write(result, "\n")
end

fptr:close()

File: HackerRank_Complete/AlternatingCharacters/AlternatingCharacters.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the alternatingCharacters function below.
int alternatingCharacters(string s) {

    int deletions = 0;

    for (int i=0; i<s.length(); i++) {
        if (i != s.length()-1 && s[i] == s[i+1]) {
            deletions++;
        }
    }

    return deletions;

}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int q;
    cin >> q;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int q_itr = 0; q_itr < q; q_itr++) {
        string s;
        getline(cin, s);

        int result = alternatingCharacters(s);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}
